build_runner/lib/src/asset_graph/node.dart

```diff
@@ -27,8 +27,34 @@ abstract class AssetNode {
   /// exist.
   Digest lastKnownDigest;
 
+  /// Whether or not this node was an output of this build.
+  bool get isGenerated => false;
+
+  /// Whether or not this asset can be read.
+  bool get isReadable => true;
+
+  /// Whether or not this node can be used as a primary or secondary input.
```

Comment:
Expand this, mention that some internal assets can be read but can't be used as primary inputs.

Reply:
done